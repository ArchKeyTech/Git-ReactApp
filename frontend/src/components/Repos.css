/*styling repo div*/
.repo-div {
    margin-top: 20px;
}
/* The heading div in the repo block */
.repo-title {
    background-color: rgba(238, 166, 12, 0.5);
    color: white;
    height: fit-content;
    margin-bottom: 0px;
    padding-bottom: 0px;
}
/* The heading styleing */
.repo-title p{
    padding: 10px;
    margin: 0px;
}

/* Styling for the repos description */
.repo-description{
    color: white;
    font-size: 14px;
}

.repo-commit-title{
    background-color: rgba(170, 170, 170, 0.089);
    height: fit-content;
}
.repo-commit-title p{
    background-color: rgba(209, 159, 22, 0.2);
    padding: 5px 0px 5px 10px;
    margin: 0px;
    font-size: 80%;
    color: white;
}

/* styling for the creation date */
.repo-date p{
    color: white;
    font-weight: bold;
    font-size: 10px;
    text-align: right;
    margin: 0px;
    padding: 2px 10px ;
}

/* Styling for the commits */
.commit-msg-div {
    padding: 10px 0px 20px 20px;
}

/* list styling */
ul {
    margin-left: 10px;
}
/*styling of repositories main title*/
.repos-main-title{
    text-align: center;
}