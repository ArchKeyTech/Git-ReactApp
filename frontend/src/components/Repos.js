import React from "react";
import "./Repos.css";

//function to display the repositories for a user in a VCS
function Repos(props) {
  //passing the repoContent state as props
  const { repoContent } = props;

  /**
   * this displays the repo content incase user has repos.
   * content includes title, description, date and commits
   */
  if ("error" in repoContent[0]) {
    return <div>{repoContent[0].error}</div>;
  } else {
    return repoContent.map((repo, key) => {
      return (
        <div key={key}>
          <div className="repo-div">
            <div className="repo-title">
              <p className="repo-name">{repo.repoName}</p>
              <p>
                {" "}
                <span className="repo-description"> {repo.repoDesc} </span>
              </p>
              <div className="repo-date">
                <p>Created on: {repo.repoBirth}</p>
              </div>
            </div>

            <div className="repo-commit-title">
              <p>Commit Feed</p>
            </div>

            <div className="commit-msg-div">
              <ul>
                {repo.commitMsg.map((msg, key) => (
                  <li className="commit-msg" key={key}>
                    {msg}
                  </li>
                ))}
              </ul>
            </div>
          </div>
        </div>
      );
    });
  }

  //returns either 'no repo' text or the repo content
  //return <div>{"error" in repoContent[0] ? repoContent[0].error : repos}</div>;
}

export default Repos;
