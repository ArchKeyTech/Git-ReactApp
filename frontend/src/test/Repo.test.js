import React from "react";
import renderer from "react-test-renderer";
import Repos from "../components/Repos";

// repo sample to mimic actual repo response
const reposSample = [
  {
    repoName: "name1",
    repoDesc: "description1",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "name2",
    repoDesc: "description2",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "name3",
    repoDesc: "description3",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "name4",
    repoDesc: "description4",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
  {
    repoName: "name5",
    repoDesc: "description5",
    createdOn: "dd-mm-yy",
    commitMsg: ["msg1", "msg2", "msg3", "msg4", "msg5"],
  },
];

// Empty repo sample to mimic when user has no repository
const reposEmptySample = [{ error: "User has no repository" }];

//test for when a complete repo content is returned as response
test("renders repo correctly", () => {
  const tree = renderer.create(<Repos repoContent={reposSample} />).toJSON();
  expect(tree).toMatchSnapshot();
});

//test for when an empty repo content is returned as response
test("renders empty repo correctly", () => {
  const tree = renderer
    .create(<Repos repoContent={reposEmptySample} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
