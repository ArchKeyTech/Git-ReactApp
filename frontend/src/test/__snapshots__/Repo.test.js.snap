// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty repo correctly 1`] = `
<div>
  User has no repository
</div>
`;

exports[`renders repo correctly 1`] = `
Array [
  <div>
    <div
      className="repo-div"
    >
      <div
        className="repo-title"
      >
        <p
          className="repo-name"
        >
          name1
        </p>
        <p>
           
          <span
            className="repo-description"
          >
             
            description1
             
          </span>
        </p>
        <div
          className="repo-date"
        >
          <p>
            Created on: 
          </p>
        </div>
      </div>
      <div
        className="repo-commit-title"
      >
        <p>
          Commit Feed
        </p>
      </div>
      <div
        className="commit-msg-div"
      >
        <ul>
          <li
            className="commit-msg"
          >
            msg1
          </li>
          <li
            className="commit-msg"
          >
            msg2
          </li>
          <li
            className="commit-msg"
          >
            msg3
          </li>
          <li
            className="commit-msg"
          >
            msg4
          </li>
          <li
            className="commit-msg"
          >
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="repo-div"
    >
      <div
        className="repo-title"
      >
        <p
          className="repo-name"
        >
          name2
        </p>
        <p>
           
          <span
            className="repo-description"
          >
             
            description2
             
          </span>
        </p>
        <div
          className="repo-date"
        >
          <p>
            Created on: 
          </p>
        </div>
      </div>
      <div
        className="repo-commit-title"
      >
        <p>
          Commit Feed
        </p>
      </div>
      <div
        className="commit-msg-div"
      >
        <ul>
          <li
            className="commit-msg"
          >
            msg1
          </li>
          <li
            className="commit-msg"
          >
            msg2
          </li>
          <li
            className="commit-msg"
          >
            msg3
          </li>
          <li
            className="commit-msg"
          >
            msg4
          </li>
          <li
            className="commit-msg"
          >
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="repo-div"
    >
      <div
        className="repo-title"
      >
        <p
          className="repo-name"
        >
          name3
        </p>
        <p>
           
          <span
            className="repo-description"
          >
             
            description3
             
          </span>
        </p>
        <div
          className="repo-date"
        >
          <p>
            Created on: 
          </p>
        </div>
      </div>
      <div
        className="repo-commit-title"
      >
        <p>
          Commit Feed
        </p>
      </div>
      <div
        className="commit-msg-div"
      >
        <ul>
          <li
            className="commit-msg"
          >
            msg1
          </li>
          <li
            className="commit-msg"
          >
            msg2
          </li>
          <li
            className="commit-msg"
          >
            msg3
          </li>
          <li
            className="commit-msg"
          >
            msg4
          </li>
          <li
            className="commit-msg"
          >
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="repo-div"
    >
      <div
        className="repo-title"
      >
        <p
          className="repo-name"
        >
          name4
        </p>
        <p>
           
          <span
            className="repo-description"
          >
             
            description4
             
          </span>
        </p>
        <div
          className="repo-date"
        >
          <p>
            Created on: 
          </p>
        </div>
      </div>
      <div
        className="repo-commit-title"
      >
        <p>
          Commit Feed
        </p>
      </div>
      <div
        className="commit-msg-div"
      >
        <ul>
          <li
            className="commit-msg"
          >
            msg1
          </li>
          <li
            className="commit-msg"
          >
            msg2
          </li>
          <li
            className="commit-msg"
          >
            msg3
          </li>
          <li
            className="commit-msg"
          >
            msg4
          </li>
          <li
            className="commit-msg"
          >
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="repo-div"
    >
      <div
        className="repo-title"
      >
        <p
          className="repo-name"
        >
          name5
        </p>
        <p>
           
          <span
            className="repo-description"
          >
             
            description5
             
          </span>
        </p>
        <div
          className="repo-date"
        >
          <p>
            Created on: 
          </p>
        </div>
      </div>
      <div
        className="repo-commit-title"
      >
        <p>
          Commit Feed
        </p>
      </div>
      <div
        className="commit-msg-div"
      >
        <ul>
          <li
            className="commit-msg"
          >
            msg1
          </li>
          <li
            className="commit-msg"
          >
            msg2
          </li>
          <li
            className="commit-msg"
          >
            msg3
          </li>
          <li
            className="commit-msg"
          >
            msg4
          </li>
          <li
            className="commit-msg"
          >
            msg5
          </li>
        </ul>
      </div>
    </div>
  </div>,
]
`;
